PWGHF/TableProducer/HFCandidateCreatorX.cxx

```diff
@@ -190,7 +197,20 @@ struct HFCandidateCreatorX {
           auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixPCA, phi, theta));
           auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixPCA, phi, 0.));
 
-          int hfFlag = 1 << XToJpsiPiPi;
+          int hfFlag = 0;
+          if (jpsiCand.hfflag() & 1 << JpsiToMuMu) {
+            hfFlag = 1 << XToJpsiToMuMuPiPi; // dimuon channel
+          } else {
+            hfFlag = 1 << XToJpsiPiPi; // dielectron channel
+          }
```

Comment:
- In the 2 prong creator this situation should be avoided for J/psi, so I have done the same for the X

Reply:
>     * In the 2 prong creator this situation should be avoided for J/psi, so I have done the same for the X

Where do you see such a switch in the 2-prong candidate creator?

Reply:
Indeed looking at your followng comment this first part is useless. I would keep only this one:
if (jpsiCand.isSelJpsiToEE() > 0) {
   hfFlag = 1 << XToJpsiPiPi;
}

if (jpsiCand.isSelJpsiToMuMu() > 0) {
   hfFlag = 1 << XToJpsiToMuMuPiPi;
}

Reply:
Well, it depends what is the intended meaning of `hfFlag`. In the 2-prong candidate creator, it is the flag from the preselection stage where all possible decay channels are considered. I am not sure what selection is supposed to be stored here.

Reply:
In this selection only the decay channel is checked while in the following one:

if (jpsiCand.isSelJpsiToEE() > 0) {
    hfFlag = 1 << XToJpsiPiPi;
}

if (jpsiCand.isSelJpsiToMuMu() > 0) {
    hfFlag = 1 << XToJpsiToMuMuPiPi;
 }

The full topological and PID selection is applied (https://github.com/AliceO2Group/O2Physics/blob/10bdabdfcde4a3890621de04068747b38961b5fd/PWGHF/TableProducer/HFJpsiCandidateSelector.cxx#L205). This second selection is embedded in "Filter filterSelectCandidates", therefore it could be redundant.

Reply:
OK, but you don't seem to consider the possibility that a candidate would pass selection for both decay channels.

Reply:
In this case I can modify it removing the "else" and putting an "if"

Reply:
Thanks for the suggestion! I will chenge the current version to:
if (jpsiCand.hfflag() & 1 << hf_cand_prong2::DecayType::JpsiToMuMu) {
 SETBIT(hfFlag, hf_cand_x::DecayType::XToJpsiToMuMuPiPi); // dimuon channel
} else {
  SETBIT(hfFlag, hf_cand_x::DecayType::XToJpsiToEEPiPi); // dielectron channel
}

Reply:
Yes, I have to replace the else. I change the if as you are suggesting!